backend/wayland/seat.c

```diff
@@ -235,11 +235,12 @@ static void keyboard_handle_leave(void *data, struct wl_keyboard *wl_keyboard,
 
 	uint32_t time = get_current_time_msec();
 
-	uint32_t pressed[dev->keyboard->num_keycodes + 1];
+	size_t num_keycodes = dev->keyboard->num_keycodes;
+	uint32_t pressed[num_keycodes];
```

Comment:
VLAs have undefined behaviour when initialized with a zero length. We should either add +1 here, or assert that `num_keycodes > 0`.

Reply:
Ah, I had forgotten that. I imagine num_keycodes can be zero, so let's just keep the +1 on the array length as it's inconsequential.